cpp/tests/trajectory/test_trajectory_distances_and_speeds.cu

```diff
@@ -14,137 +14,97 @@
-
-  auto id = sorted->get_column(0);
-  auto xs = sorted->get_column(1);
-  auto ys = sorted->get_column(2);
-  auto ts = sorted->get_column(3);
-
-  auto distance_and_speed =
-    cuspatial::trajectory_distances_and_speeds(offsets->size(), id, xs, ys, ts, this->mr());
-
-  using Rep = typename cudf::timestamp_ms::rep;
-
-  auto h_xs      = cudf::test::to_host<T>(xs).first;
-  auto h_ys      = cudf::test::to_host<T>(ys).first;
-  auto h_ts      = cudf::test::to_host<Rep>(ts).first;
-  auto h_offsets = cudf::test::to_host<int32_t>(*offsets).first;
-
-  std::vector<double> speed{};
-  std::vector<double> distance{};
-
-  // compute expected distance and speed
-  std::tie(distance, speed) = expected_distance_and_speed<T, Rep>(h_offsets, h_xs, h_ys, h_ts);
-
-  auto speed_actual    = distance_and_speed->get_column(1);
-  auto distance_actual = distance_and_speed->get_column(0);
-
-  auto speed_expected = cudf::test::fixed_width_column_wrapper<double>(speed.begin(), speed.end());
-  auto distance_expected =
-    cudf::test::fixed_width_column_wrapper<double>(distance.begin(), distance.end());
-
-  cudf::test::expect_columns_equivalent(distance_actual, distance_expected);
-  cudf::test::expect_columns_equivalent(speed_actual, speed_expected);
+  auto const size = 1000;
+
+  {
+    auto id =
+      cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));  // not integer
+    auto xs = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ys = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ts = cudf::column(rmm::device_uvector<cudf::timestamp_ms>(size, rmm::cuda_stream_default));
+    EXPECT_THROW(cuspatial::trajectory_distances_and_speeds(1, id, xs, ys, ts, this->mr()),
+                 cuspatial::logic_error);
+  }
+  {
+    auto id = cudf::column(rmm::device_uvector<int>(size, rmm::cuda_stream_default));
+    auto xs = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ys = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ts =
+      cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));  // not timestamp
+    EXPECT_THROW(cuspatial::trajectory_distances_and_speeds(1, id, xs, ys, ts, this->mr()),
+                 cuspatial::logic_error);
+  }
+  {
+    // x-y type mismatch
+    auto id = cudf::column(rmm::device_uvector<cudf::size_type>(size, rmm::cuda_stream_default));
+    auto xs = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ys = cudf::column(rmm::device_uvector<double>(size, rmm::cuda_stream_default));
+    auto ts = cudf::column(rmm::device_uvector<cudf::timestamp_ms>(size, rmm::cuda_stream_default));
+    EXPECT_THROW(cuspatial::trajectory_distances_and_speeds(1, id, xs, ys, ts, this->mr()),
+                 cuspatial::logic_error);
+  }
 }
 
-TYPED_TEST(TrajectoryDistanceSpeedTest, ComputeDistanceAndSpeed3Simple)
```

Comment:
There are a few hand derived tests in original test sets. Should we try to reintroduce them in the tests for header only API?

Reply:
I felt they were redundant to the new tests.